cmd/run/run.go

```diff
@@ -436,7 +458,14 @@ func run(sims []*Simulation, bind simulator.BindAddr, size int) error {
 		}
 		sim.Cleanup()
 	}
-
-	printGoodbye()
+	// Extract list of failed simulations and return as an error.
+	failedSimNames := failedSimulationNames(failedSims)
+	printGoodbye(failedSimNames != nil)
+	if failedSimNames != nil {
+		msg := fmt.Sprintf(
+			"The following simulations experienced errors: %v",
+			strings.Join(failedSimNames, ", "))
+		return errors.New(msg)
+	}
```

Comment:
`printGoodbye` is a function which is called only once, so the only reason for its existence is to make the code cleaner. Splitting half of the goodbye message into `printGoodbye` and leaving another half inlined in the function doesn't make sense. We should either pass the failed simulations into the printGoodbye or remove printGoodbye and inline everything.